acp/DPI.md

```diff
@@ -0,0 +1,45 @@
+## DefraDB Policy Interface (DPI) Restrictions
+To be a DPI compliant resource, these rules MUST be met:
+- The resource must exist.
+- The resource must atleast have the required registerer relation under `relations`.
+- The resource must atleast have all the required permissions under `permissions`.
+- Every required permission must have the required registerer relation in `expr`.
+- The required registerer relation must be the leading (first) relation in `expr` (see example below).
+
+To be a 'DPI' policy, all of it's resources must be DPI compliant.
+To be 'Partially-DPI' atleast one of it's resource must be DPI compliant.
+
+Learn more about DefraDB ACP Terminologies [TERMINOLOGY](/acp/TERMINOLOGY.md)
+
+### Moreover:
+All the required permissions are:
+    - Defined in the `dpi.go` file in variable `dpiRequiredPermissions`.
+The required registerer relation name is:
+    - Defined in the `dpi.go` file in variable `requiredRegistererRelationName`.
+
+## DPI Resource Examples:
+- Check out tests here: [tests/integration/acp/schema/add_dpi](/tests/integration/acp/schema/add_dpi)
+- The tests linked are broken into `accept_*_test.go` and `reject_*_test.go` files.
+- Accepted tests document the valid DPIs (as the schema is accepted).
+- Rejected tests document invalid DPIs (as the schema is rejected).
+- There are also some Partially-DPI tests that are both accepted and rejected depending on the resource.
+
+### DPI Resource's Required Permission's Expression:
+Eventhough the following expressions are valid generic policy expressions, they will make a
+DPI compliant resource loose it's DPI status as these expressions are not in accordance to
+our DPI rules defined above. Assuming these `expr` are under a required permission label:
+- `expr: ownerMalicious + owner`
```

Comment:
suggestion: Given that `DPI` is a distinct thing, and `owner` is always required in a DPI expression, why to we want to force users to manually specify `owner` (and understand and get write and read documentation explaining it) instead of auto-magically substitute it in?

Reply:
By we do you mean DefraDB or SourceHub?

Reply:
DefraDB

Reply:
Are you suggesting DefraDB parses and edits the user's given policy before it is handed off / uploaded to sourcehub?

Reply:
yes

Reply:
That smells like a pain to manage on defraDB side and encouraging a lack of transparency between the users.

Even if we choose to go down the route of manipulating and editing users policies, I think it should be through sourcehub (maybe with metapolicies) as it already will make the parsed AST.

I am more in favor of helping them write correct policies, i.e. through some auto-filled templates and better documentation, or even auto-generating policies for them based on them answering a couple of questions related to the data ownership model their app needs.

Reply:
Can bring up in standup and follow in a thread if needs be

Reply:
Discussed in standup today, everyone is agreed on this being out of scope atm.